api/net/Areas/Subscriber/Controllers/ReportInstanceController.cs

```diff
@@ -243,6 +243,9 @@ public async Task<IActionResult> PublishAsync(int id)
         var instance = _reportInstanceService.FindById(id) ?? throw new NoContentException("Report does not exist");
         if (instance.OwnerId != user.Id) throw new NotAuthorizedException("Not authorized to publish this report"); // User does not own the report
 
+        instance.Status = Entities.ReportStatus.Submitted;
```

Comment:
Change report instance state to submitted immediately so that the UI can lock down.